tidy3d/components/data/sim_data.py

```diff
@@ -1115,11 +1115,21 @@ def process_adjoint_sources(self, adj_srcs: list[SourceType]) -> AdjointSourceIn
             hashes_to_src_times[tmp_src_hash].append(src.source_time)
 
         num_ports = len(hashes_to_src_times)
-        num_unique_freqs = len({src.source_time.freq0 for src in adj_srcs})
+        unique_freqs = {src.source_time.freq0 for src in adj_srcs}
+        num_unique_freqs = len(unique_freqs)
 
         # next, figure out which treatment / normalization to apply
         if num_unique_freqs == 1:
             log.info("Adjoint source creation: one unique frequency, no normalization.")
+            freqs_adj = self.simulation.freqs_adjoint
+
+            # if many adjoint freqs, but only 1 unique, need to mask out the non-contributors
+            if len(freqs_adj) > 1:
+                coords = dict(f=freqs_adj)
+                data = [1 if f == tuple(unique_freqs)[0] else 0 for f in freqs_adj]
+                post_norm = xr.DataArray(data, coords=coords)
+                return AdjointSourceInfo(sources=adj_srcs, post_norm=post_norm, normalize_sim=True)
+
```

Comment:
Unrelated to this PR, but this is not in `pre/2.8`, we should probably do another merge from develop? We probably shouldn't be doing that merge together with this PR.